### PR TITLE
Update prometheus federator app version to 0.3.5 / charts to 0.4.1

### PR DESCRIPTION
**Pull request by alexandreLamarre**
Related Issue: 

### Checklist

Please fill out this table to identify which fields need to be modified in your PR.

**Under `Status`, either indicate `Does Not Apply` or `Added to this PR`**.

| **Version to be incremented**                                              | Why should this be modified?                                                                                                           | Status                                      |
|-----------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------|
| `version` in rancher-project-monitoring `package.yaml`                  | You modified the contents of the `rancher-project-monitoring` chart to make changes                                                    | Added in this PR  |
| `helmProjectOperator.image.tag` in prometheus-federator `values.yaml` | Either you modified the rancher-project-monitoring chart or you modified the `main.go` file                                          | Does Not Apply |
| `appVersion` in prometheus-federator `Chart.yaml`                     | You modified the `helmProjectOperator.image.tag` in the above box                                                                      | Added in this PR |
| `version` in prometheus-federator `Chart.yaml`                        | Either you modified the `appVersion` in the above box or you modified the contents of the `prometheus-federator` chart to make changes | Added in this PR |

### CODE REVIEW

File: packages/prometheus-federator/charts/values.yaml [hunk 1]
```diff
@@ -58,7 +58,7 @@ helmProjectOperator:
 
   image:
     repository: rancher/prometheus-federator
-    tag: v0.3.4
+    tag: v0.3.5
```
Comment (joshmeranda) [inline packages/prometheus-federator/charts/values.yaml:61]: I don't think this tag exists?

File: packages/prometheus-federator/charts/Chart.yaml [hunk 2]
```diff
@@ -7,12 +7,12 @@ annotations:
   catalog.cattle.io/provides-gvr: helm.cattle.io.projecthelmchart/v1alpha1
   catalog.cattle.io/release-name: prometheus-federator
 apiVersion: v2
-appVersion: 0.3.4
+appVersion: 0.3.5
```
Comment (joshmeranda) [inline packages/prometheus-federator/charts/Chart.yaml:10]: we aren't changing anything in the actual federator so this should stay the same.

**Review by joshmeranda — CHANGES_REQUESTED** [review]
See comments, just a versioning issue

File: packages/prometheus-federator/charts/Chart.yaml (hunk 2 above)
Comment (alexandreLamarre) [reply packages/prometheus-federator/charts/Chart.yaml:10]: We have to change this to build the new image with go 1.20

File: packages/prometheus-federator/charts/values.yaml (hunk 1 above)
Comment (alexandreLamarre) [reply packages/prometheus-federator/charts/values.yaml:61]: In the docs it mentions adding the git tag after a PR is merged, which didn't seem right but looking at how the automation works it seems that its fine

File: packages/prometheus-federator/charts/Chart.yaml (hunk 2 above)
Comment (joshmeranda) [reply packages/prometheus-federator/charts/Chart.yaml:10]: Oh ok sick

File: packages/prometheus-federator/charts/values.yaml (hunk 1 above)
Comment (alexandreLamarre) [reply packages/prometheus-federator/charts/values.yaml:61]: Taking a look at past PRs, we are expected to tag the merge commit as the release version, CI runs with the a local $TAG option

**Review by joshmeranda — APPROVED** [review]

**Pull request merged**